docs/src/main/asciidoc/using-vertx.adoc

```diff
@@ -0,0 +1,334 @@
+            .map(l -> String.format("Hello %s! (%s)%n", name, new Date()));
+}
+----
+
+The second approach slightly differs:
+
+[source,java]
+----
+// Use io.vertx.reactivex.core.Vertx;
+@Inject Vertx vertx;
+
+@GET
+@Produces(MediaType.SERVER_SENT_EVENTS)
+@Path("{name}/streaming")
+public Publisher<String> greeting(@PathParam("name") String name) {
+    return vertx.periodicStream(2000).toFlowable()
+            .map(l -> String.format("Hello %s! (%s)%n", name, new Date()));
+}
+----
+
+The server side is ready.
+In order to see the result in the browser, we need a web page.
+
+[source,html]
+.META-INF/resources/streaming.html
+----
+<!doctype html>
+<html>
+<head>
+    <meta charset="utf-8"/>
+    <title>SSE with Vert.x - Quarkus</title>
+    <script type="application/javascript" src="streaming.js"></script>
+</head>
+<body>
+<div id="container"></div>
+</body>
+</html>
+----
+
+Our web page just has an empty `<div>` container.
+The magic, as always, lies in the Javascript code:
+
+[source,javascript]
+.META-INF/resources/streaming.js
+----
+var eventSource = new EventSource("/hello/Quarkus/streaming");
+eventSource.onmessage = function (event) {
+    var container = document.getElementById("container");
+    var paragraph = document.createElement("p");
+    paragraph.innerHTML = event.data;
+    container.appendChild(paragraph);
+};
+----
+
+IMPORTANT: Most browsers support SSE but some don't.
+More about this in  Mozilla's https://developer.mozilla.org/en-US/docs/Web/API/EventSource#Browser_compatibility[SSE browser-compatibility list].
+
+Navigate to http://localhost:8080/streaming.html.
+A new greeting should show-up every 2 seconds.
+
+[source, text]
+----
+Hello Quarkus! (Thu Mar 21 17:26:12 CET 2019)
+
+Hello Quarkus! (Thu Mar 21 17:26:14 CET 2019)
+
+Hello Quarkus! (Thu Mar 21 17:26:16 CET 2019)
+
+...
+----
+
+=== Using JSON results
+
+Vert.x API heavily relies on JSON, namely the `io.vertx.core.json.JsonObject` and `io.vertx.core.json.JsonArray` types.
+They are both supported as {project-name} web resource return types.
+
+Consider these endpoints:
+
+[source,java]
+----
+@Path("/hello")
+@Produces(MediaType.APPLICATION_JSON)
+public class VertxJsonResource {
+
+    @GET
+    @Path("{name}/object")
+    public JsonObject jsonObject(@PathParam("name") String name) {
+        return new JsonObject().put("Hello", name);
+    }
+
+    @GET
+    @Path("{name}/array")
+    public JsonArray jsonArray(@PathParam("name") String name) {
+        return new JsonArray().add("Hello").add(name);
+    }
+}
+----
+
+In your browser, navigate to http://localhost:8080/hello/Quarkus/object. You should see:
+
+[source, text]
+----
+{"Hello":"Quarkus"}
+----
+
+Then, navigate to http://localhost:8080/hello/Quarkus/array:
+
+[source, text]
+----
+["Hello","Quarkus"]
+----
+
+Needless to say, this works equally well when the JSON result is wrapped in a `CompletionStage` or a `Publisher`.
+
+== Using the event bus
```

Comment:
How about renaming this section to "Going further", or "Beyond the basics"... whatever.

Then below we can link to different quides:
- async message passing (in the making)
- web client (in the making)
- pg client
- ...

Reply:
I was thinking about having a whole section about clients. We also need to cover the `@Route`.

Reply:
Not sure what you mean? You convinced me it was a good idea to split the content. Where do you want to add the client section?

Reply:
I've fixed this, and you are right it's already long enough. I've added a small section about using clients.